machine1/dbus.go

```diff
@@ -75,6 +77,75 @@ func (c *Conn) initConnection() error {
 	return nil
 }
 
+// GetImage may be used to get the image object path for the image with the specified name
+func (c *Conn) GetImage(name string) (dbus.ObjectPath, error) {
+	var path dbus.ObjectPath
+	result := c.object.Call(dbusInterface+".GetImage", 0, name)
+	if result.Err != nil {
+		log.Print(result.Err)
+		return path, result.Err
+	}
+	pathstr, ok := result.Body[0].(dbus.ObjectPath)
+	if ok {
```

Comment:
Generally we prefer a) short-circuit returns, and b) explicitly returning `(nil, err)` in error cases - which also simplifies the code a bit:

```
func (c *Conn) GetImage(name string) (dbus.ObjectPath, error) {
	result := c.object.Call(dbusInterface+".GetImage", 0, name)
	if result.Err != nil {
		return nil, result.Err
	}
	path, ok := result.Body[0].(dbus.ObjectPath)
	if !ok {
		return nil, fmt.Errorf("Error converting '%v' to dbus.ObjectPath.", result.Body[0])
 	}
 	return path, nil
 }
```